include/pybind11/class_support.h

```diff
@@ -173,16 +144,36 @@ inline PyTypeObject* make_default_metaclass() {
     return type;
 }
 
-/// Instance creation function for all pybind11 types. It only allocates space for the
-/// C++ object, but doesn't call the constructor -- an `__init__` function must do that.
-extern "C" inline PyObject *pybind11_object_new(PyTypeObject *type, PyObject *, PyObject *) {
+/** Instance creation function for all pybind11 types. It only allocates space for the C++ object
+ * (or multiple objects, for Python-side inheritance from multiple pybind11 types), but doesn't call
+ * the constructor -- an `__init__` function must do that.
+ */
+extern "C" PYBIND11_NOINLINE inline PyObject *pybind11_object_new(PyTypeObject *type, PyObject *, PyObject *) {
```

Comment:
Right; that's a leftover of an earlier implementation detail (that did call it) that I ended up reverting; I'll remove it.